### PR TITLE
Collection CRUD Methods

### PR DESCRIPTION
**Pull request by gagelarsen**
- [x] FileCollectionClient.delete()
- [x] FileCollectionClient.duplicate()
- [x] FileCollectionClient.export()

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ if [ ! -f "$1" ]; then
 fi
 rm -f .coverage
 echo "Running Unit Tests..."
-coverage run -a --rcfile=coverage.ini -m unittest -v tethysext.atcore.tests.unit_tests
+#coverage run -a --rcfile=coverage.ini -m unittest -v tethysext.atcore.tests.unit_tests
```
Comment (swainn) [inline test.sh]: You know what to do...

File: tethysext/atcore/models/file_database/file_collection_client.py [hunk 2]
```diff
@@ -48,20 +53,75 @@ def new(cls, session: Session, file_database_id: uuid.UUID, meta: dict = None) -
...
+        Property to get the underlying instance so it can be lazy loaded.
+
+        Returns:
+            The underlying FileCollection instance.
+
+        Raises:
+            Exception: If the the instance has been deleted.
+        """
+        if self.__deleted:
+            raise Exception('The collection has been deleted.')
```
Comment (swainn) [inline tethysext/atcore/models/file_database/file_collection_client.py:66]: Create a custom exception: UnboundClient

File: tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py [hunk 3]
```diff
@@ -263,3 +276,45 @@ def test_set_meta_new_value(self):
...
+        self.assertTrue(os.path.exists(collection_path))
+        collection_client.delete()
+        self.assertFalse(os.path.exists(collection_path))
+
+        with self.assertRaises(Exception) as exc:
+            _ = collection_client.path
+
+        self.assertTrue('The collection has been deleted.' in str(exc.exception))
+
+    def test_collection_export(self):
```
Comment (swainn) [inline tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py:296]: Add a second case to copy nested directories and files.

File: tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py [hunk 4]
```diff
@@ -263,3 +276,45 @@ def test_set_meta_new_value(self):
...
+        collection_path = os.path.join(root_dir, str(database_client.instance.id), str(collection_client.instance.id))
+        export_path = os.path.join(self.test_files_base, 'temp', 'exported_files', 'test_collection_export')
+        if os.path.exists(export_path):
+            shutil.rmtree(export_path)
+        self.assertTrue(os.path.exists(collection_path))
+        collection_client.export(export_path)
+        self.assertTrue(os.path.exists(export_path))
+        self.assertTrue(os.path.exists(os.path.join(export_path, '__meta__.json')))
+
+    def test_collection_duplicate(self):
```
Comment (swainn) [inline tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py:311]: Add a second case to copy nested directories and files.

**Review by swainn — CHANGES_REQUESTED** [review]

File: test.sh (hunk 1 above)
Comment (gagelarsen) [reply test.sh]: done

**Review by swainn — APPROVED** [review]

**Pull request merged**